src/main/java/edu/cornell/kfs/vnd/document/CuVendorMaintainableImpl.java

```diff
@@ -225,4 +229,84 @@ private void setupMultiselectField(MaintenanceDocument document, MaintenanceDocu
         }
         
     }
+    
+    @Override
+    public void doRouteStatusChange(DocumentHeader header) {
+        LOG.debug("doRouteStatusChange: entering");
+        super.doRouteStatusChange(header);
+        VendorDetail vendorDetail = (VendorDetail) getBusinessObject();
+        WorkflowDocument workflowDoc = header.getWorkflowDocument();
+        
+        if (workflowDoc.isProcessed()) {
+            LOG.debug("doRouteStatusChange: workflow is processed");
+            if (recordPvenRouteStatusChangeForPaymentWorksIsEnabled()) {
+                performPaymentWorksApprovalProcessingForVendor(vendorDetail, this.getDocumentNumber(), this.getMaintenanceAction());
+            }
+        }
+        else if (workflowDoc.isDisapproved() || workflowDoc.isCanceled()) {
+            LOG.debug("doRouteStatusChange: disapproved or canceled");
+            if (recordPvenRouteStatusChangeForPaymentWorksIsEnabled()) {
+                performPaymentWorksDisapprovalCancelProcessingForVendor(vendorDetail, this.getDocumentNumber(), this.getMaintenanceAction());
+            }
+        }
+    }
+    
+    private void performPaymentWorksApprovalProcessingForVendor(VendorDetail vendorDetail, String kfsDocumentNumber, String maintenanceAction) {
+        if (isExistingPaymentWorksVendor(kfsDocumentNumber) && isMaintenanceActionNewOrNewWithExisting(maintenanceAction)) {
+            LOG.debug("performPaymentWorksApprovalProcessingForVendor: isExistingPaymentWorksVendor");
+            this.saveBusinessObject();
+            processExistingPaymentWorksVendorApproval(kfsDocumentNumber, vendorDetail);
+        }
+        else if (isMaintenanceActionNewOrNewWithExisting(maintenanceAction)) {
+            LOG.debug("performPaymentWorksApprovalProcessingForVendor: new action resulting from AVf or hand keying");
+            this.saveBusinessObject();
+            processKfsVendorNewActionApproval(kfsDocumentNumber, vendorDetail);
+        }
+        else if (StringUtils.equals(KRADConstants.MAINTENANCE_EDIT_ACTION, maintenanceAction)) {
+            LOG.debug("performPaymentWorksApprovalProcessingForVendor: edit action");
+            processKfsVendorEditActionApproval(kfsDocumentNumber, vendorDetail);
+        }
```

Comment:
Is a plain "else" block needed here to at least log the cases where no PaymentWorks processing is needed?

Reply:
Good suggestion. I have added as an info message since we would want to see it in the log when it is encountered and this condition should not be hit much so it should not generate a lot of noise.